src/modules/gateways/callback/paymentwall.php

```diff
@@ -16,30 +16,35 @@
 
 require_once(ROOTDIR . "/includes/api/paymentwall_api/lib/paymentwall.php");
 
-$gateway = getGatewayVariables("paymentwall");
+$invoiceid = $_GET['goodsid'];
+if(!$invoiceid) {
+    die('Order is invalid!');
+}
+
+$orderData = mysql_fetch_assoc(select_query('tblorders', 'userid,id,paymentmethod', array("invoiceid" => $invoiceid)));
+
+$gateway = getGatewayVariables($orderData['paymentmethod']);
 
 if (!$gateway["type"]) {
-    die("Module Not Activated");
+    die($gateway['name'] . " is not activated");
 }
 
+$pingback = new Paymentwall_Pingback($_GET, $_SERVER['REMOTE_ADDR']);

```

Comment:
Please create a function named `getRealClientIp` to support request from proxies like `CloudFlare` or `nginx`